Tests/Editor/CoreApi/ManagedEC2DeploymentTests.cs

```diff
@@ -0,0 +1,201 @@
+            _coreApiMock.Setup(f => f.GetSetting(It.IsAny<SettingsKeys>())).Returns(Response.Ok(new GetSettingResponse()));
+
+            _managedEc2FleetParametersMock.Object.GameName = GameName;
+            _managedEc2FleetParametersMock.Object.FleetName = FleetName;
+            _managedEc2FleetParametersMock.Object.BuildName = BuildName;
+            _managedEc2FleetParametersMock.Object.GameServerFile = GameServerFile;
+            _managedEc2FleetParametersMock.Object.GameServerFolder = GameServerFolder;
+            _managedEc2FleetParametersMock.Object.OperatingSystem = OperatingSystemName;
+            
+            const string testScenarioName = "test scenario";
+            const string testScenarioDescription = "test scenario description";
+            const string testScenarioUrl = "test";
+
+            Mock<ScenarioLocator> scenarioLocatorMock = SetUpScenarioLocatorToReturnTestDeployer(
+                testScenarioName, testScenarioDescription, testScenarioUrl, hasServer: false, _coreApiMock);
+
+            _deploymentSettings = GetUnitUnderTest(scenarioLocator: scenarioLocatorMock);
+            
+            _deploymentSettings.Refresh();
+            return new ManagedEC2Deployment(_deploymentSettings, _managedEc2FleetParametersMock.Object);
+        }
+        
+        [Test]
+        public void StartDeployment_WhenCorrectInputs_ExpectSuccess()
+        {
+            //Arrange
+            var ec2DeploymentHappyPath = ArrangeEc2DeploymentHappyPath();
+            
+            //Act
+            ec2DeploymentHappyPath.StartDeployment();
+            
+            //Assert
+            _coreApiMock.Verify(f => f.PutSetting(It.IsAny<SettingsKeys>(), It.IsAny<string>()), Times.Exactly(8));
+            
+            Assert.AreEqual(_deploymentSettings.GameName, GameName);
+        }
+        
+        [Test]
+        public void StartDeployment_WhenTaskIsFaulted_ExpectException()
+        {
+            //Arrange
+            var ec2DeploymentHappyPath = ArrangeEc2DeploymentHappyPath();
+            
+            //Act & Assert
+            Assert.Throws<AggregateException>(() => _deploymentSettings.StartDeployment(null).Wait());
+        }
+        
+        [Test]
+        public void StartDeployment_WhenTaskIsFaultedWithContinueWith_ExpectFalse()
+        {
+            //Arrange
+            var ec2DeploymentHappyPath = ArrangeEc2DeploymentHappyPath();
+            var isErrored = false;
+            
+            //Act
+            _deploymentSettings.StartDeployment(null).ContinueWith(task =>
+            {
+                if (task.IsFaulted)
+                {
+                    isErrored = true;
+                }
+            });
+                
+            //Assert
+            Assert.IsFalse(isErrored);
+        }
+        
+        [Test]
+        public void DeleteDeployment_WhenCorrectInputs_ExpectSuccess()
```

Comment:
I'd argue there's no reason to have a unit test for a delegate method (a method that only calls an identical method in a separate instance). Potentially this test could be moved to DeploymentSettingsTests (this case might be tested already so no need to add it again)